packages/loopring_v3/contracts/impl/libexchange/ExchangeGenesis.sol

```diff
@@ -62,4 +73,41 @@ library ExchangeGenesis
         S.registerToken(address(0));
         S.registerToken(loopring.lrcAddress());
     }
+
+    function increaseAccountCapacity(
+        ExchangeData.State storage S
+        )
+        external
+    {
+        require(S.accountTreeDepth < MAX_ACCOUNT_TREE_DEPTH, "MAX_TREE_SIZE_REACHED");
+        S.accountTreeDepth += 1;
+
+        S.merkleRoot = hashImpl(
+            S.merkleRoot, S.emptyMerkleRoot, S.emptyMerkleRoot, S.emptyMerkleRoot
+        );
+
+        S.emptyMerkleRoot = hashImpl(
+            S.emptyMerkleRoot, S.emptyMerkleRoot, S.emptyMerkleRoot, S.emptyMerkleRoot
+        );
+    }
+
+    function hashImpl(
+        bytes32 t0,
+        bytes32 t1,
+        bytes32 t2,
+        bytes32 t3
+        )
+        private
+        pure
+        returns (bytes32)
+    {
+        Poseidon.HashInputs5 memory inputs = Poseidon.HashInputs5(
```

Comment:
@Brechtpd do we need to do `>> 3` trick here?

Reply:
No, the Poseidon hashes are limited by the field size automatically.